src/main/java/org/testng/internal/Invoker.java

```diff
@@ -1309,6 +1312,18 @@ private void handleInvocationResults(
     }
   }
 
+  private IRetryAnalyzer getRetryAnalyzerConsideringMethodParameteters(ITestNGMethod tm, ITestResult tr) {
+    Object[] key = tr.getParameters();
+    IRetryAnalyzer retryAnalyzer = tm.getRetryAnalyzer();
+    if ((key != null && key.length != 0) && retryAnalyzer != null) {
+      String keyAsString = tm.getTestClass().getRealClass().getName() + "_" +
+          tm.getMethodName() + "_" + Arrays.toString(key);
+      retryAnalyzer = m_testMethodToRetryAnalyzer.computeIfAbsent(keyAsString,
+          o -> ClassHelper.newInstance(tm.getRetryAnalyzer().getClass()));
```

Comment:
If I understand well the fix, we need a new instance by test method (or test result) instance?
Does it work if you map by TestResult instance (it will simplify the code)?

I wonder if `tm.getRetryAnalyzer()` is broken and should be deprecated. Then maybe the current logic could be moved somewhere else.

Reply:
> Does it work if you map by TestResult instance (it will simplify the code)?

No it doesn't work.

TestNG already produces a new instance of `IRetryAnalyzer` per test method (`@Test` method). 

> we need a new instance by test method (or test result) instance?

What we need is a new instance of `IRetryAnalyzer` per parameter invocation for a given `@Test` method which is what this PR attempts at addressing.

The logic had to be housed in `Invoker` instead of `ITestNGMethod` because it is not aware of the parameters that are associated with it. That information is currently available only within a `ITestResult` object.

> I wonder `if tm.getRetryAnalyzer()` is broken and should be deprecated. 

I now added a new method that retrieves the `IRetryAnalyzer` from the `ITestNGMethod` when an `ITestResult` object is passed. I have amended the commit and pushed in the changes.